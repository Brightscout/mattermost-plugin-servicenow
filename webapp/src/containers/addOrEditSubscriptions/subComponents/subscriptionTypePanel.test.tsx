import React from 'react';

import {shallow, ShallowWrapper} from 'enzyme';

import SubscriptionTypePanel from './subscriptionTypePanel';
import { ModalSubtitleAndError } from 'mm-ui-library';

const mockOnContinue = jest.fn();
const mockOnBack = jest.fn();
const mockSetSubscriptionType = jest.fn();

const mockSubscriptionType: SubscriptionType = 'record';

const subscriptionTypePanelProps = {
    className: 'mockClassName',
    error: 'mockError',
    onBack: mockOnBack,
    onContinue: mockOnContinue,
    actionBtnDisabled: true,
    requiredFieldValidationErr: true,
    subscriptionType: mockSubscriptionType,
    setSubscriptionType: mockSetSubscriptionType,
};

describe('Subscription Type Panel', () => {
    let component: ShallowWrapper<any, Readonly<{}>, React.Component<{}, {}, any>>;

    beforeEach(() => {
        component = shallow(<SubscriptionTypePanel {...subscriptionTypePanelProps}/>);
    });

    afterEach(() => {
        component.unmount();
    });

    it('Should render correctly', () => {
        expect(component).toMatchSnapshot();
    });

    it('Should apply the passed className prop', () => {
        expect(component.hasClass(subscriptionTypePanelProps.className)).toBeTruthy();
    });

    it('Should render the subscription type panel body correctly', () => {
        expect(component.find('Dropdown')).toHaveLength(1);
        expect(component.find('ModalSubtitleAndError')).toHaveLength(1);
        expect(component.find('ModalFooter')).toHaveLength(1);
    });

    it('Should render the error correctly', () => {
        expect(component.contains(
            <ModalSubtitleAndError error={subscriptionTypePanelProps.error}/>,
        )).toBeTruthy();
    });

    it('Should fire change event when dropdown value is changed', () => {
        const changeDropdown = (changeNumber: number) => {
            component.find('Dropdown').simulate('change');
            expect(subscriptionTypePanelProps.setSubscriptionType).toHaveBeenCalledTimes(changeNumber);
        };

        // Click the checkbox
        changeDropdown(1);
        changeDropdown(2);
    });
});
