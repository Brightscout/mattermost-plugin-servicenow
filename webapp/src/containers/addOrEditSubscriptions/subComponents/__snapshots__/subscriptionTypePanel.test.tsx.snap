// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Subscription Type Panel Should render correctly 1`] = `
<div
  className="modal__body wizard__secondary-panel mockClassName"
>
  <div
    className="padding-h-12 padding-v-20 subscription-type-panel"
  >
    <Dropdown
      error={false}
      onChange={[Function]}
      options={
        Array [
          Object {
            "label": "Record subscription",
            "value": "record",
          },
          Object {
            "label": "Bulk subscription",
            "value": "object",
          },
        ]
      }
      placeholder="Select Subscription Type"
      required={true}
      value="record"
    />
    <ModalSubtitleAndError
      error="mockError"
    />
  </div>
  <ModalFooter
    cancelBtnText="Back"
    cancelDisabled={true}
    confirmBtnText="Continue"
    confirmDisabled={true}
    onConfirm={[Function]}
    onHide={[MockFunction]}
  />
</div>
`;
